src/lib.rs

```diff
@@ -36,3 +44,26 @@ unsafe extern "C" fn realpath(
 
     resolved_path
 }
+
+#[no_mangle]
+unsafe extern "C" fn clock_gettime(
+    clock_id: libc::clockid_t,
+    tp: *mut libc::timespec,
+) -> libc::c_int {
+    let mut retval = -1;
+    match clock_id {
+        libc::CLOCK_REALTIME => {
+            let mut tv = MaybeUninit::uninit();
+
+            retval = libc::gettimeofday(tv.as_mut_ptr(), ptr::null_mut());
+            if retval == 0 {
+                let tv = tv.assume_init();
+                (*tp).tv_nsec = (tv.tv_usec * 1000).into();
```

Comment:
This `into` isn't needed. Clippy flags this as a warning:
```
warning: useless conversion to the same type: `i64`
  --> src/lib.rs:61:33
   |
61 |                 (*tp).tv_nsec = (tv.tv_usec * 1000).into();
   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `.into()`: `(tv.tv_usec * 1000)`
   |
   = note: `#[warn(clippy::useless_conversion)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
```

```suggestion
                (*tp).tv_nsec = tv.tv_usec * 1000;
```

Reply:
Huh, that's odd, I could have sworn I had a `cargo check` failure when I wrote that... maybe I hadn't set the target properly yet and they had two different types.

Reply:
Yeah `cargo check` and `cargo clippy` are two different things